### ISSUE TITLE
System report template "set reportSystem..." refers incorrectly?

### ISSUE THREAD

**Issue by BarWeinig** [EDCD/EDDI#2175]
## Note:
I have spent the evening, together with my brother, testing and double-checking. This is a total re-write of this report to remove as many variables as possible.

## What's Wrong:
TL:DR : It seems impossible, with the currently set variables to get System power information, once arrived in that system.

### Expected
When running the `System report` template, once arrived in a system, EDDI should be able to tell about the system power. 

### Observed
The default `System report` template includes a script to tell about the system's power. But I'm very sure now, that script will never trigger, once in the system.

### Steps to reproduce
To delete all possible variables, I have rewritten the 2 scripts I use that involve the system's power. In both cases I have normalized the naming to camelCase. The only things these scripts now do, is tell us the value of the variables, based on the different `reportSystem`s by default available in these 2 templates. 

Replace the `FSD engaged` template with:
```
{_ Everything in context, set as normal _}
{SetState('eddi_context_last_subject', 'fsd')}
{SetState('eddi_context_last_action', 'engage')}
{SetState('eddi_context_system_name', event.system)}
{SetState('eddi_context_system_system', event.system)}
{SetState('eddi_context_fsd_target', event.target)}

{if event.target = 'Hyperspace':
    {SetState('eddi_context_last_subject', 'jump')}
    {SetState('eddi_context_last_action', 'start')}

	{_ This will work! _}
    {set reportSystem to SystemDetails(event.system)}

    {Pause(5000)}

    {_ start of check _}
    Now using the event dot system:
	System is {P(reportSystem.name)},
	the powers name is {P(reportSystem.power, "power")} and
	invariant name is {reportSystem.powerState.invariantName}.
}
```
Replace the `System report` template with:
```
{_ Normal setting of reportSystem _}
{if state.eddi_context_system_name:
    {set reportSystem to SystemDetails(state.eddi_context_system_name)}
|else:
    {set reportSystem to SystemDetails(system.systemname)}
}

{_ Re-setting for bug-report purposes _}
{set reportSystem to SystemDetails(state.eddi_context_system_name)}
Now using the state dot eddi context system name:
System is {P(reportSystem.name)},
the powers name is {P(reportSystem.power, "power")} and
invariant name is {reportSystem.powerState.invariantName}.

{set reportSystem to SystemDetails(system.systemname)}
Now using the system dot system name:
System is {P(reportSystem.name)},
the powers name is {P(reportSystem.power, "power")} and
invariant name is {reportSystem.powerState.invariantName}.
```
I added some comments to clarify. Now, with these templates, jump to a system that has a power and is not contested. Once arrived in the system, ask EDDI for a system report using VoiceAttack. The result will be as displayed under EDDI Logs. 

### Configuration
  * EDDI Version: 4.0.0-b3
  * VoiceAttack Version: v1.8.7
* Localization [if applicable]
  * EDDI's language: English United States
  * Voice: Not of interest

## My Investigation
See above

### Investigation Notes
See above

### EDDI Logs
For `FSD engaged`:
```
Now using the event dot system: System is Khered, the powers name is Pranav Antal and invariant name is Exploited.
```
For `System report`:
```
Now using the state dot eddi context system name: System is Khered, the powers name is and invariant name is None. 
Now using the system dot system name: System is Khered, the powers name is and invariant name is None.
```

As you can see, it works in the former, in the latter the powers name is "" and the invariant-name is "None".

### Player journals
Seems not needed. If you'd like, I can do this.

**Comment by JibstaMan** [comment]
Congrats on opening your first GitHub issue ever! 😄

**Comment by BarWeinig** [comment]
> 
> 
> Congrats on opening your first GitHub issue ever! 😄

Thanks Bro ;)

**Comment by Tkael** [comment]
`FSD engaged` changes the current system. I'm not sure when you're triggering your system report? Are you using it as a child script to `FSD engaged`?

I don't have your full scripts, but there also may be a casing issue that could lead to a difference in behavior between the two scripts... `reportsystem` in `FSD engaged` vs. `reportSystem` in `System report`. Cottle variables are case sensitive.

Testing using the script below and the systems you mentioned in your report...

```
{reportSystem.systemname} is controlled by {reportSystem.power} and is in the {reportSystem.powerState.invariantName} state.
```

- Khered is controlled by Pranav Antal and is in the Exploited state
- Lalande 2450 is controlled by None and is in the state.

Dropping your script snippet in for Khered, I get `, and is aligned to Pranav Antal.`

Testing a few other notable systems...

- Sol is controlled by Zachary Hudson and is in the Control state.
- Achenar is controlled by Arissa Lavigny-Duval and is in the Exploited state.
- Alioth is controlled by Edmund Mahon and is in the Control state.

Note that `{reportSystem.powerState.invariantName}` is ___not set___ when there is no power. Since an empty variable is not equal to "None", your modified script equates not having any data with the "is free from associated power" state rather than the "inconclusive about the systems relation to the powers" state. This explains the result you saw with Lalande 2450. 

You aren't getting the expected result for Khered either though (from either script), which makes me also suspect you might have had a connectivity issue to the server where we draw powerplay data. Logs would be helpful to see if there were any connection issues logged.

**Comment by BarWeinig** [comment]
The entire issue-report has been rewritten, based on more research and the above response.

**Comment by BarWeinig** [comment]
> You aren't getting the expected result for Khered either though (from either script), which makes me also suspect you might have had a connectivity issue to the server where we draw powerplay data. Logs would be helpful to see if there were any connection issues logged.

I have now spent yet another day, just trying to work around this issue. And I think I did that. But, now looking back at this issue-report, it still includes information that is not needed. The only thing you need to do to encounter this issue is use the System report function in VoiceAttack. In a system with a power (contested or not) you will never hear EDDI tell you the correct information. It's data will always show: "" for powername and "None" for powerstate. You could use the script I provided above, to get rid of a lot of information that does exist and works as intended, but it is not needed. 

Sidenote: I have now used a test template to test when EDDI does know Powerplay information and it seems to be, mainly later on in the jump procedure. Right after the countdown for Hyperspace has finished, EDDI has a slim chance of already having Powerplay information (downloaded). Once in the system all that information is gone and EDDI doesn't know Powerplay information (unless you set variables to remember what EDDI knew in Hyperspace).

TL:DR : Get EDDI to tell you Powerplay information, using the System report function in VoiceAttack and this bug is fixed. That part has nothing to do with internetspeeds, nor the FSD engaged template...

**Comment by Tkael** [comment]
Ah ha! I think I've found it. 
Essentially, there seems to be a bug where jumping into a system overwrites the power data from server with power data from the journal... even if you aren't pledged and the journal is therefore reporting no power data.
